src/sentry/tasks/deletion.py

```diff
@@ -56,12 +73,10 @@ def run_deletion(deletion_id):
     if deletion.aborted:
         raise DeleteAborted
 
-    if not deletion.in_progress:
+    if first_pass:
         actor = deletion.get_actor()
         instance = deletion.get_instance()
-        with transaction.atomic():
-            deletion.update(in_progress=True)
-            pending_delete.send(sender=type(instance), instance=instance, actor=actor)
+        pending_delete.send(sender=type(instance), instance=instance, actor=actor)
```

Comment:
This signal only has one receiver right now for repositories. I'm pretty sure that receiver is never triggered right now and will confirm that before tackling repository deletion.